user/signer.go

```diff
@@ -0,0 +1,50 @@
+package user
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	neofsecdsa "github.com/nspcc-dev/neofs-sdk-go/crypto/ecdsa"
+)
+
+// Signer is an interface of entities that can be used for signing operations
+// in NeoFS. It is the same as [neofscrypto.Signer], but has an extra method to retrieve [ID].
+type Signer interface {
+	neofscrypto.Signer
+
+	UserID() ID
+}
+
+// SignerRFC6979 wraps [ecdsa.PrivateKey] and represents signer based on deterministic
+// ECDSA with SHA-256 hashing (RFC 6979). Provides [Signer] interface.
+//
+// Instances SHOULD be initialized with [NewSignerRFC6979] or [NewSignerRFC6979WithID].
+type SignerRFC6979 struct {
```

Comment:
they are literally named the same (`user.SignerRFC6979` and `neofsecdsa.SignerRFC6979`). was it discussed too? @cthulhu-rider, @roman-khimov

Reply:
Is it a problem?

Reply:
well, one is the "original" while the other one is a wrapper over the first one. also `user.Signer*` confuses me (but `*ecdsa.Signer*` is clear). also, it reminds me of the problem of reading some neofs-node code when it is not clear what struct is here: sdk or api-go

Reply:
Any alternative?

Reply:
as i understand, you have already discussed it so i am only able to suggest naming with `User`, `Wrapper`, `WithID`. all of them are bad if the whole concept is taken as the only way